packages/react-devtools-scheduling-profiler/src/types.js

```diff
@@ -123,6 +123,22 @@ export type FlamechartStackLayer = FlamechartStackFrame[];
 
 export type Flamechart = FlamechartStackLayer[];
 
+export type HorizontalScrollStateChangeCallback = (
+  scrollState: ScrollState,
+) => void;
+
+// Imperative view state that corresponds to profiler data.
```

Comment:
just curious, why does this new state live outside of React lifecycle?

Reply:
It already did.